esmvaltool/cmorizers/obs/cmorize_obs_glodap.py

```diff
@@ -0,0 +1,130 @@
+"""ESMValTool CMORizer for GLODAP data.
+
+Tier
+   Tier 2: other freely-available dataset.
+
+Source
+   GLODAP: https://www.glodap.info/index.php/mapped-data-product/
+
+Last access
+   GLODAP: 20210528
+
+Download and processing instructions
+   All handled by the script (download only if local raw data are missing)
+
+Modification history
+   20210528-lovato_tomas: written
+
+"""
+
+import logging
+import os
+from warnings import catch_warnings, filterwarnings
+import tarfile
+import requests
+import iris
+from cf_units import Unit
+
+from .utilities import (constant_metadata, fix_coords, fix_var_metadata,
+                        save_variable, set_global_atts)
+
+logger = logging.getLogger(__name__)
+
+
+def _fix_data(cube, var):
+    """Specific data fixes for different variables."""
+    logger.info("Fixing data ...")
+    with constant_metadata(cube):
+        if var in [
+                'dissic',
+                'talk',
+        ]:
+            cube /= 1000.  # Convert from umol/kg to mol/m^3
+    return cube
+
+
+def collect_files(in_dir, var, cfg):
+    """Compose input file list and download if missing."""
+    var_dict = cfg['variables'][var]
+
+    fname = '.'.join([var_dict['file'], var_dict['raw_var'], 'nc'])
+    in_file = os.path.join(in_dir, fname)
+
+    # check if input file is missing
+    if not os.path.isfile(in_file):
+        if not os.path.isdir(in_dir):
+            os.makedirs(in_dir)
+
+        # check if raw tar file is in place
+        tar_file = os.path.basename(cfg['attributes']['source'])
+        tar_file = os.path.join(in_dir, tar_file)
+        if not os.path.isfile(tar_file):
+            logger.info('Input file %s is missing. Start download ... ',
+                        tar_file)
+            url_file = requests.get(cfg['attributes']['source'])
+            open(tar_file, 'wb').write(url_file.content)
+
+        # get input file from tar archive
+        tar_file = tarfile.open(name=tar_file, mode='r')
+        tar_base = os.path.basename(cfg['attributes']['source'])[:-7]
```

Comment:
can you make this a bit more forwards-compatible? what if file name changes?

Reply:
@valeriupredoi I think that if the filename is changed, data will be modified as well (these netcdf are really messy) so the cmorizer will need further adjustments ... probably we can survive with this little name tailoring ...

Reply:
I assume that this can be solved ...

Reply:
yep! cheers for the explanation :+1: